java/maven/src/org/netbeans/modules/maven/execute/MavenCommandLineExecutor.java

```diff
@@ -1048,24 +1046,21 @@ private File checkAvailability(String ver, VersionRange vr, InputOutput ioput) {
                     LOGGER.log(Level.WARNING, "wasn''t able to download maven binaries, version {0}", ver);
                     return null;
                 }
-                str = new ZipInputStream(is);
-                ZipEntry entry;
-                while ((entry = str.getNextEntry()) != null) {
-                    //base it of f not child as the zip contains the maven base folder
-                    File fileOrDir = new File(f,  entry.getName());
-                    if (entry.isDirectory()) {
-                        fileOrDir.mkdirs();
-                    } else {
-                        FileOutputStream fos = null;
-                        try {
-                            fos = new FileOutputStream(fileOrDir);
-                            FileUtil.copy(str, fos);
-                        } finally {
-                            IOUtil.close(fos);
-                        }
-                        // correct way to set executable flag?
-                        if ("bin".equals(fileOrDir.getParentFile().getName()) && !fileOrDir.getName().endsWith(".conf")) {
-                            fileOrDir.setExecutable(true);
+                try (ZipInputStream str = new ZipInputStream(is)) {
+                    ZipEntry entry;
+                    while ((entry = str.getNextEntry()) != null) {
+                        //base it of f not child as the zip contains the maven base folder
+                        File fileOrDir = new File(f,  entry.getName());
+                        if (entry.isDirectory()) {
+                            fileOrDir.mkdirs();
+                        } else {
+                            try (FileOutputStream fos = new FileOutputStream(fileOrDir)) {
+                                FileUtil.copy(str, fos);
+                            }
```

Comment:
could be replaced with:
```java
   Files.copy(str, fileOrDir.toPath())
```
`fileOrDir` could be potentially converted to path from the beginning and use `Files.createDirectories` etc.


or:
```java
in.transferTo(out);
```
what you prefer :)

Reply:
`FileUtil.copy` -> `Files.copy()` will have another cleanup PR. Good catch!